libraries/chain/combined_database.cpp

```diff
@@ -4,19 +4,20 @@
 
 namespace eosio { namespace chain {
 
-   combined_session::combined_session(chainbase::database& cb_database)
-       : cb_session{ std::make_unique<chainbase::database::session>(cb_database.start_undo_session(true)) } {}
-
-   combined_session::combined_session(chainbase::database& cb_database, b1::chain_kv::undo_stack& kv_undo_stack)
-       : kv_undo_stack{ &kv_undo_stack } {
-      try {
+   combined_session::combined_session(chainbase::database& cb_database, b1::chain_kv::undo_stack* kv_undo_stack)
+       : kv_undo_stack{ kv_undo_stack } {
+      if (!kv_undo_stack) {
+         cb_session = std::make_unique<chainbase::database::session>(cb_database.start_undo_session(true));
```

Comment:
I'm guessing I am missing something, but how come you don't just initialize cb_session (not in an if check), and then just do if(kv_undo_stack) { ... push(false) ... }

Reply:
That is so that if an exception is thrown from the cb_session constructor, the application is terminated by analogy with other methods. Should I make this a special case?

Reply:
I guess do this for now, it passed by @larryk85 before.